src/dmd/backend/tk.d

```diff
@@ -0,0 +1,20 @@
+/**
+ * Compiler implementation of the
+ * $(LINK2 http://www.dlang.org, D programming language).
+ *
+ * Copyright:   Copyright (C) 1999-2019 by The D Language Foundation, All Rights Reserved
+ * Authors:     $(LINK2 http://www.digitalmars.com, Walter Bright)
+ * License:     $(LINK2 http://www.boost.org/LICENSE_1_0.txt, Boost License 1.0)
+ * Source:      $(LINK2 https://github.com/dlang/dmd/blob/master/src/dmd/backend/tk.c, backend/tk.c)
+ */
+
+import core.stdc.stdlib : malloc, calloc, realloc, free;
+import core.stdc.string : strdup;
+
+extern (C) @nogc nothrow:
+char* mem_strdup(const char* p) { return strdup(p); }
+void* mem_malloc(size_t u) { return malloc(u); }
```

Comment:
This was my first thought too.  I discussed in with @WalterBright in a private e-mail.

Me:  mem.c seems to just be a wrapper around malloc and friends. Would it be alright to just replace all calls to `mem_malloc`, `mem_free`, etc. to `malloc`, `free`, etc. respectively?
@WalterBright: Convert mem.c to D, don't replace the calls.

@WalterBright, might you reconsider accepting this PR.  It would probably be easier to rewrite the debug code and alternative allocators in `tk.c` and `mem.c` from scratch in D, than it would be to convert them from C.  Please consider accepting this PR.

Reply:
I suspect that Walter's response meant: follow the general porting guideline - port the code as close as possible, keeping the same behavior as before and only after that start refactoring. This PR should remove the uncertainty.